internal/registry/config.go

```diff
@@ -4,47 +4,87 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"strconv"
 	"strings"
 
+	"github.com/kyma-project/cli.v3/internal/kube"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
 	"k8s.io/client-go/kubernetes"
 )
 
 const (
-	RegistrySecretName  = "serverless-registry-config-default"
-	serverlessNamespace = "kyma-system"
+	LegacyRegistrySecretName  = "serverless-registry-config-default"
+	LegacyServerlessNamespace = "kyma-system"
 )
 
 type RegistryConfig struct {
+	SecretName string
+	SecretData *SecretData
+	PodMeta    *RegistryPodMeta
+}
+
+func GetConfig(ctx context.Context, client kube.Client) (*RegistryConfig, error) {
+	var secretConfig *SecretData
+	var secretName string
+
+	dockerRegistry, err := getServedDockerRegistry(ctx, client.Dynamic())
```

Comment:
IMO this part of code is hard to read. Whole method is long.

Reply:
done